cmd/lambda/populate_reset_queue/main.go

```diff
@@ -1,80 +1,85 @@
 package main
 
 import (
-	"encoding/json"
 	"log"
 
-	"github.com/pkg/errors"
-
-	"github.com/Optum/dce/pkg/common"
-	"github.com/Optum/dce/pkg/db"
+	"github.com/Optum/dce/pkg/account"
+	"github.com/Optum/dce/pkg/config"
+	"github.com/Optum/dce/pkg/event/eventiface"
 	"github.com/aws/aws-lambda-go/events"
 	"github.com/aws/aws-lambda-go/lambda"
-	"github.com/aws/aws-sdk-go/aws/session"
-	"github.com/aws/aws-sdk-go/service/sqs"
 )
 
-// addAccountToQueue publishes a single account ID into the SQS
-// as an event for consumption
-func addAccountToQueue(accounts []*db.Account, queueURL *string,
-	queue common.Queue, dbSvc db.DBer) error {
-	// For each Account, send the message to Reset Queue and update
-	// FinanceLock Lease status if necessary
-	for _, acct := range accounts {
-		// Send Message
-		body, err := json.Marshal(acct)
-		if err != nil {
-			return errors.Wrapf(err, "Failed to add account %s to queue accounts", acct.ID)
-		}
-		sBody := string(body)
-		err = queue.SendMessage(queueURL, &sBody)
-		if err != nil {
-			return errors.Wrapf(err, "Failed to add account %s to queue accounts", acct.ID)
-		}
-		log.Printf("%s : Added to Reset Queue\n", acct.ID)
-	}
-	return nil
-
+type configuration struct {
+	Debug         string `env:"DEBUG" envDefault:"false"`
+	ResetQueueURL string `env:"RESET_SQS_URL" envDefault:"SqsUrl"`
 }
 
-// rbenqHandler is the base handler function for the lambda
-func rbenqHandler(cloudWatchEvent events.CloudWatchEvent) error {
+var (
+	services *config.ServiceBuilder
+	// Settings - the configuration settings for the controller
+	settings *configuration
+)
 
-	// Create Database Service
-	dbSvc, err := db.NewFromEnv()
-	if err != nil {
-		return err
+func init() {
+	cfgBldr := &config.ConfigurationBuilder{}
+	settings = &configuration{}
+	if err := cfgBldr.Unmarshal(settings); err != nil {
+		log.Fatalf("Could not load configuration: %s", err.Error())
 	}
 
-	// Get NotReady Accounts
-	accounts, err := dbSvc.FindAccountsByStatus(db.NotReady)
+	// load up the values into the various settings...
+	err := cfgBldr.WithEnv("AWS_CURRENT_REGION", "AWS_CURRENT_REGION", "us-east-1").Build()
 	if err != nil {
-		log.Printf("Failed to list accounts: %s\n", err)
-		return err
+		log.Printf("Error: %+v", err)
 	}
+	svcBldr := &config.ServiceBuilder{Config: cfgBldr}
 
-	// Create the Queue Service
-	queueURL := common.RequireEnv("RESET_SQS_URL")
-	awsSession, err := session.NewSession()
+	_, err = svcBldr.
+		WithAccountService().
+		Build()
 	if err != nil {
-		return err
+		panic(err)
+	}
+
+	services = svcBldr
+
+}
+
+// Handler is the base handler function for the lambda
+func Handler(cloudWatchEvent events.CloudWatchEvent) error {
+
+	var eventSvc eventiface.Servicer
+	if err := services.Config.GetService(&eventSvc); err != nil {
```

Comment:
Didn't we do something in another controller, just like:

```golang
eventSvc := service.GetEventer()
```

that's a nicer pattern, IMO, if we can do it.

Reply:
Yea I think we just need to decide how far down that rabbit hole we want to go.  When do we decide if a service is going to have the easier pattern.